src/Time/StepChoosers/ElementSizeCfl.hpp

```diff
@@ -0,0 +1,105 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <pup.h>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Time/StepChoosers/StepChooser.hpp"
+#include "Domain/SizeOfElement.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class GlobalCache;
+}  // namespace Parallel
+namespace domain {
+namespace Tags {
+template <size_t Dim>
+struct SizeOfElement;
+}  // namespace Tags
+}  // namespace domain
+// IWYU pragma: no_forward_declare db::DataBox
+/// \endcond
+
+namespace StepChoosers {
+/// Suggests a step size based on the CFL stability criterion, but uses the full
+/// size of the element as the length scale in question.
+///
+/// This is useful as a coarse estimate for slabs, or to place a ceiling on
+/// another dynamically-adjusted step chooser.
+template <typename StepChooserUse, size_t Dim, typename System>
+class ElementSizeCfl : public StepChooser<StepChooserUse> {
+ public:
+  /// \cond
+  ElementSizeCfl() = default;
+  explicit ElementSizeCfl(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(ElementSizeCfl);  // NOLINT
+  /// \endcond
+
+  struct SafetyFactor {
+    using type = double;
+    static constexpr Options::String help{"Multiplier for computed step"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  static constexpr Options::String help{
+      "Suggests a step size based on the CFL stability criterion, but in which "
+      "the entire size of the element is used as the spacing in the "
+      "computation. This is useful primarily for placing a ceiling on another "
+      "dynamically-adjusted step chooser"};
+  using options = tmpl::list<SafetyFactor>;
+
+  explicit ElementSizeCfl(const double safety_factor) noexcept
+      : safety_factor_(safety_factor) {}
+
+  using argument_tags =
+      tmpl::list<::Tags::TimeStepper<>, domain::Tags::SizeOfElement<Dim>,
+                 typename System::compute_largest_characteristic_speed>;
+  using return_tags = tmpl::list<>;
+  using compute_tags =
+      tmpl::list<domain::Tags::SizeOfElementCompute<Dim>,
```

Comment:
how does adding the compute tag work for events? I think currently `Initialization::Actions::Minmod` is adding `SizeOfElementCompute`. It would be good to avoid any annoying coupling between the step choosers and the limiter code

Reply:
I think currently the compute tags for the step choosers are currently added via a `AddComputeTags` action in initialization. I don't think it should cause any problems for another action to also list the same compute tag in the list -- the `SetupDataBox` should remove duplicates, so if I recall correctly, the first listed one should be the one that is kept. It would be a problem if there were two different tags that shared a base tag, but I don't think that can happen for this case as things stand currently.
Does that answer the concern?

Reply:
Yep, thank you!